src/base/bittorrent/session.cpp

```diff
@@ -68,12 +68,6 @@
 #include <QTimer>
 #include <QUuid>
 
-#ifdef Q_OS_WIN
-// TODO: Remove together with fixBrokenSavePath()
```

Comment:
Any idea what will happen to users who upgrade from 4.2.3 to a later version that does not include this fix? How can we identify/deal with bug reports that result from that?

Reply:
4.2.3 is completely unusable for the users affected by this bug. I'm really sure none of them use it anymore.

Reply:
@glassez 

Ok, but imagine the case where:

- Casual user had 4.2.3, paths broke, did not update since
- User updates to some version post-4.2.5 that includes this change in this PR.
- Paths are not fixed in the fastresume, user runs into bugs.

I realize it may be a bit unlikely, but still.

Reply:
>I realize it may be a bit unlikely, but still.

IMO, there was plenty of time to take advantage of this fix. Otherwise, we must keep all this deprecated/upgrade/compatibility code forever. I would not like to return it unless absolutely necessary.

Reply:
How about putting a version specific check in place ie must be 4.2.3+ in installer or something.

Note could be made on official website when an official version is released with this PR that you must install at least 4.2.3+ to install this version?

Reply:
> IMO, there was plenty of time to take advantage of this fix. Otherwise, we must keep all this deprecated/upgrade/compatibility code forever. I would not like to return it unless absolutely necessary.

Of course. What I would like to know is simply: in the case I pointed out, if a user posts an issue report because of it, how easy is it to diagnose the problem correctly as "the broken paths bug" and tell the user to roll back to 4.2.5 and then upgrade back to the new version?

> How about putting a version specific check in place ie must be 4.2.3+ in installer or something.

Or just a warning like "WARNING: if you are upgrading to 4.2.3, please upgrade to 4.2.5 first".

I think in the long term, the fastresume saving mechanism needs to overhauled to prevent backwards/forwards compatibility issues like these. Or at least ensure that changes in the formats are always smooth across newer versions such as to not potentially force the user to go through the rigmarole of upgrading from A to B before installing C.

Reply:
All your suggestions about require the user to update via some particular intermediate version looks too overkill for me. Then it's really simpler to keep this "fixing" code.
But asking once again. If some affected user did neither update qBittorrent to release that fixes this bug nor fix it in a different way within past 4 months should we care about?

Reply:
>But asking once again. If some affected user did neither update qBittorrent to release that fixes this bug nor fix it in a different way within past 4 months should we care about?

Code-wise, I'd say **NO** & continue PR as normal.

Would still put an advisory disclaimer on main website upon it's official release as a suggestion rather than a requirement so as to avoid any new issue reports "popping up" as per @FranciscoPombal's use case above.

^This can be discussed/decided prior to/at (next official release time)

Reply:
@glassez 

> If some affected user did neither update qBittorrent to release that fixes this bug nor fix it in a different way within past 4 months should we care about?

Echoing xavier's thoughts, code-wise, no.

But we do care about them when they come here and post a non-reproducible bug report, because at some point we will all forget that this happened in 4.2.3 and will be confused as to why certain users are having save path issues with the new version. Perhaps it's too much of a pessimistic anticipation, though (I expect only a vanishingly small minority of users to be in this situation of upgrading directly from 4.2.3 ->4.2.6/4.3.x), and even a a partial solution for this can be accomplished outside of this PR (e.g. an announcement/warning on the site/installer/whatever).

So as far as code changes go, this is fine by me I guess.

Reply:
I was asking in terms of code changes.
I also think it's a good idea to publish information about known bugs and related things on the site.